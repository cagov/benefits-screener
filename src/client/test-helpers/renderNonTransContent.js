import React from "react";
import { shallow } from "enzyme";
import { I18nextProvider, Trans } from "react-i18next";
import i18n from "./i18nForTests";
import shallowUntilTarget from "./shallowUntilTarget";

function renderNonTransContent(Component, props = {}) {
  const providersWrapper = shallow(
    <I18nextProvider i18n={i18n}>
      <Component {...props} />
    </I18nextProvider>
  );

  const wrapper = shallowUntilTarget(providersWrapper, Component.name);
  // Incorrectly using renderNonTransContent to render a component that
  // contains a Trans component results in a silent failure (the rendered
  // snapshot will contain the Trans component itself instead of the rendered
  // contents). This test makes that silent failure more obvious.
  expect(wrapper.containsMatchingElement(<Trans />)).toBe(false);

  return wrapper;
}

module.exports = renderNonTransContent;
