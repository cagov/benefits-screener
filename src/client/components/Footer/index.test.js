import renderNonTransContent from "../../test-helpers/renderNonTransContent";
import Component from "./index";

describe("<Footer />", () => {
  it("renders the component", async () => {
    const wrapper = renderNonTransContent(Component, "Footer");

    expect(wrapper).toMatchSnapshot();
  });
});
