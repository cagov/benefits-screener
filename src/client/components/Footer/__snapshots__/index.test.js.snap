// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders the component 1`] = `
<footer
  className="footer"
>
  <Navbar
    bg="dark"
    className="justify-content-between"
    collapseOnSelect={false}
    expand={true}
    variant="custom"
  >
    <Nav
      className="flex-wrap"
      fill={false}
      justify={false}
    >
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        disabled={false}
        href="/#back-to-top"
      >
        Back to Top
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        disabled={false}
        href="https://edd.ca.gov/about_edd/"
        rel="noopener noreferrer"
        target="_blank"
      >
        About EDD
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        disabled={false}
        href="https://edd.ca.gov/about_edd/contact_edd.htm"
        rel="noopener noreferrer"
        target="_blank"
      >
        Contact EDD
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        disabled={false}
        href="https://edd.ca.gov/about_edd/conditions_of_use.htm"
        rel="noopener noreferrer"
        target="_blank"
      >
        Conditions of Use
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        disabled={false}
        href="https://edd.ca.gov/about_edd/privacy_policy.htm"
        rel="noopener noreferrer"
        target="_blank"
      >
        footer.privacyPolicy
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        disabled={false}
        href="https://edd.ca.gov/about_edd/accessibility.htm"
        rel="noopener noreferrer"
        target="_blank"
      >
        Accessibility
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        disabled={false}
        href="https://edd.ca.gov/sitemap.htm"
        rel="noopener noreferrer"
        target="_blank"
      >
        Site Map
      </NavLink>
    </Nav>
    <Nav
      className="flex-wrap"
      fill={false}
      justify={false}
    >
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        disabled={false}
        href="https://www.facebook.com/californiaedd/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Facebook icon"
          height="15"
          src="images/facebook.svg"
          width="15"
        />
         
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        disabled={false}
        href="https://twitter.com/CA_EDD"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Twitter icon"
          height="15"
          src="images/twitter.svg"
          width="15"
        />
         
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        disabled={false}
        href="https://www.linkedin.com/company/californiaedd/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="LinkedIn icon"
          height="15"
          src="images/linkedin.svg"
          width="15"
        />
         
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        disabled={false}
        href="https://www.instagram.com/ca_edd/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Instagram icon"
          height="15"
          src="images/instagram.svg"
          width="15"
        />
         
      </NavLink>
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        disabled={false}
        href="https://www.youtube.com/user/CaliforniaEDD"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="YouTube icon"
          height="15"
          src="images/youtube.svg"
          width="15"
        />
         
      </NavLink>
    </Nav>
  </Navbar>
  <div
    className="bg-light secondary-footer"
  >
    <Container
      fluid={false}
    >
      Copyright Â© 2020 State of California
    </Container>
  </div>
</footer>
`;
